src/platform/tls_openssl.c

```diff
@@ -202,16 +202,29 @@ CxPlatTlsAlpnSelectCallback(
     return SSL_TLSEXT_ERR_OK;
 }
 
+BOOLEAN
+CxPlatTlsVerifyCertificate(
+    _In_ X509* X509Cert,
+    _In_ const char* SNI
+    );
+
 static
 int
 CxPlatTlsCertificateVerifyCallback(
     int preverify_ok,
     X509_STORE_CTX *x509_ctx
     )
 {
+    X509* Cert = X509_STORE_CTX_get0_cert(x509_ctx);
     SSL *Ssl = X509_STORE_CTX_get_ex_data(x509_ctx, SSL_get_ex_data_X509_STORE_CTX_idx());
     CXPLAT_TLS* TlsContext = SSL_get_app_data(Ssl);
 
+    if (!preverify_ok) {
```

Comment:
Is there every a case where if openssl says the certificate is valid where we still want to check the windows store?